CromIAM/src/main/scala/cromiam/webservice/CromIamApiService.scala

```diff
@@ -185,29 +191,41 @@ trait CromIamApiService extends RequestSupport
                                              action: String,
                                              request: HttpRequest,
                                              cromwellAuthClient: CromwellClient,
-                                             cromwellRequestClient: CromwellClient): Future[HttpResponse] = {
-    def authForCollection(collection: Collection): Future[Unit] = {
-      samClient.requestAuth(CollectionAuthorizationRequest(user, collection, action), request) recoverWith {
-        case e: SamDenialException => Future.failed(e)
-        case e =>
-          log.error(e, "Unable to connect to Sam {}", e)
-          Future.failed(SamConnectionFailure("authorization", e))
-      }
+                                             cromwellRequestClient: CromwellClient):
+  FailureResponseOrT[HttpResponse] = {
+    def authForCollection(collection: Collection): FailureResponseOrT[Unit] = {
```

Comment:
```scala
      samClient.
        requestAuth(CollectionAuthorizationRequest(user, collection, action), request).
        mapK{
          Lambda[FunctionK[IO, IO]] {
            _ handleErrorWith {
              case e: SamDenialException => IO.raiseError(e)
              case e =>
                log.error(e, "Unable to connect to Sam {}", e)
                IO.raiseError(SamConnectionFailure("authorization", e))
            }
          }
        }
    }

```

requires kind projector.  Again not necessarily better but when I see `.value` i.e. going in and out of the transformer I start to wonder

Reply:
Found a way that compiled without using non's kind-projector, using `FunctionK.lift()`